httpcore/_async/http2.py

```diff
@@ -265,16 +276,21 @@ async def request(
             b"content-length" in seen_headers or b"transfer-encoding" in seen_headers
         )
 
-        await self.send_headers(method, url, headers, has_body, timeout)
-        if has_body:
-            await self.send_body(stream, timeout)
-
-        # Receive the response.
-        status_code, headers = await self.receive_response(timeout)
-        reason_phrase = get_reason_phrase(status_code)
-        stream = AsyncByteStream(
-            aiterator=self.body_iter(timeout), aclose_func=self._response_closed
-        )
+        await self.connection.max_streams_semaphore.acquire()
```

Comment:
I'm not really familiar with the mechanics of HTTP/2 but it occurs to me that maybe we ought to not block if no more streams are available but rather create a new connection?

That might be a separate issue though.

Reply:
Different issue - this is about the maximum number of *concurrent* streams, rather than the maximum stream id before a new connection should be reissued.

Reply:
Right, makes sense 👍